pkg/operator/resourcesynccontroller/interfaces.go

```diff
@@ -4,6 +4,7 @@ package resourcesynccontroller
 type ResourceLocation struct {
 	Namespace string `json:"namespace"`
 	Name      string `json:"name"`
+	Optional  bool   `json:"optional"`
```

Comment:
`Required`.  requiring this is exceptional, not normal.

Reply:
ok